Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift

```diff
@@ -739,17 +717,16 @@ AppDelegateViewModelOutputs {
     return applicationDidFinishLaunchingReturnValueProperty.value
   }
 
-  fileprivate let notificationAuthorizationCompletedProperty = MutableProperty()
-  public func notificationAuthorizationCompleted() {
-    self.notificationAuthorizationCompletedProperty.value = ()
+  fileprivate let notificationAuthorizationCompletedProperty = MutableProperty(false)
+  public func notificationAuthorizationCompleted(isGranted: Bool) {
+    self.notificationAuthorizationCompletedProperty.value = isGranted
   }
 
   fileprivate let notificationAuthorizationStatusProperty =
     MutableProperty<NotificationAuthorizationStatusType?>(nil)
-  public func notificationAuthorizationStatusReceived(
-    _ notificationSetting: NotificationAuthorizationStatusType
-  ) {
-    return self.notificationAuthorizationStatusProperty.value = notificationSetting
+  @available(iOS 10.0, *)
+  public func notificationAuthorizationStatusReceived(_ authorizationStatus: UNAuthorizationStatus) {
+    return self.notificationAuthorizationStatusProperty.value = authStatusType(for: authorizationStatus)
```

Comment:
Here is my solution to avoid the protocol extension: convert the `UNAuthorizationStatus` right here with a lil function. How do you all feel about this?